transcendence-app/src/user/user.service.ts

```diff
@@ -1,27 +1,45 @@
 import { Injectable } from '@nestjs/common';
 import { UserDto } from './dto/user.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserEntity } from '../typeorm/user.entity';
 
 @Injectable()
 export class UserService {
+  constructor(
+    @InjectRepository(UserEntity)
+    private readonly userRepository: Repository<UserEntity>,
+  ) {}
+
   private readonly users: UserDto[] = [];
 
-  private findOne(username: string) {
-    const user = this.users.find((user) => user.username === username);
-    return user;
+  async findOne(username: string) {
+    return await this.userRepository.findOne({ username: username });
```

Comment:
We could shortcircuit to a null as gbudau did. Why is null better than undefined? I don't know. I'll ask there.